includes/API/Packs/ApiLabkiPacksList.php

```diff
@@ -0,0 +1,650 @@
+
+		foreach ( $packs as $pack ) {
+			$packData = $this->buildPackData( $pack, $repo, $ref, $pageRegistry, $includePages );
+			$allPacks[] = $packData;
+		}
+
+		return $allPacks;
+	}
+
+	/**
+	 * Get a specific pack by pack ID.
+	 *
+	 * @param int $packId Pack ID
+	 * @param bool $includePages Whether to include page data
+	 * @return array Array with single pack data structure
+	 */
+	private function getSinglePackById( int $packId, bool $includePages ): array {
+		$packRegistry = $this->getPackRegistry();
+		$pageRegistry = $this->getPageRegistry();
+		$repoRegistry = $this->getRepoRegistry();
+		$refRegistry = $this->getRefRegistry();
+
+		// Get the pack
+		$pack = $packRegistry->getPack( $packId );
+		if ( $pack === null ) {
+			$this->dieWithError(
+				'labkipackmanager-error-pack-not-found',
+				'pack_not_found'
+			);
+		}
+
+		// Get the ref and repo for context
+		$ref = $refRegistry->getRefById( $pack->contentRefId() );
+		$repo = $ref ? $repoRegistry->getRepo( $ref->contentRepoId()->toInt() ) : null;
+
+		// Build and return as array
+		return [ $this->buildPackData( $pack, $repo, $ref, $pageRegistry, $includePages ) ];
+	}
+
+	/**
+	 * Get a specific pack by name.
+	 *
+	 * @param \LabkiPackManager\Domain\ContentRefId $refId Ref ID
+	 * @param string $packName Pack name
+	 * @param bool $includePages Whether to include page data
+	 * @return array Array with single pack data structure
+	 */
+	private function getSinglePackByName(
+		\LabkiPackManager\Domain\ContentRefId $refId,
+		string $packName,
+		bool $includePages
+	): array {
+		$packRegistry = $this->getPackRegistry();
+		$pageRegistry = $this->getPageRegistry();
+		$repoRegistry = $this->getRepoRegistry();
+		$refRegistry = $this->getRefRegistry();
+
+		// Get the pack by name
+		$packId = $packRegistry->getPackIdByName( $refId, $packName );
+		if ( $packId === null ) {
+			$this->dieWithError(
+				'labkipackmanager-error-pack-not-found',
+				'pack_not_found'
+			);
+		}
+
+		$pack = $packRegistry->getPack( $packId );
+		if ( $pack === null ) {
+			$this->dieWithError(
+				'labkipackmanager-error-pack-not-found',
+				'pack_not_found'
+			);
+		}
+
+		// Get the ref and repo for context
+		$ref = $refRegistry->getRefById( $refId );
+		$repo = $ref ? $repoRegistry->getRepo( $ref->contentRepoId()->toInt() ) : null;
+
+		// Build and return as array
+		return [ $this->buildPackData( $pack, $repo, $ref, $pageRegistry, $includePages ) ];
+	}
+
+	/**
+	 * Build complete pack data structure.
+	 *
+	 * @param \LabkiPackManager\Domain\Pack $pack Pack domain object
+	 * @param \LabkiPackManager\Domain\ContentRepo|null $repo Repository domain object
+	 * @param \LabkiPackManager\Domain\ContentRef|null $ref Ref domain object
+	 * @param \LabkiPackManager\Services\LabkiPageRegistry $pageRegistry Page registry
+	 * @param bool $includePages Whether to include page data
+	 * @return array Complete pack data structure
+	 */
+	private function buildPackData(
```

Comment:
this is a method for converting the `Pack` object to a serializable form, right? that should be a method on the `Pack` class like `toJson()` or something so that packs always have a canonical serialization. this risks packs being represented in inconsistent ways across api endpoints.